packages/wrangler/src/__tests__/helpers/mock-cfetch.ts

```diff
@@ -140,23 +140,23 @@ export function setMockResponse<ResponseType>(
 /**
  * A helper to make it easier to create `FetchResult` objects in tests.
  */
-export function createFetchResult<ResponseType>(
-  result: ResponseType,
+export async function createFetchResult<ResponseType>(
+  result: ResponseType | Promise<ResponseType>,
   success = true,
   errors = [],
   messages = [],
   result_info?: unknown
-): FetchResult<ResponseType> {
+): Promise<FetchResult<ResponseType>> {
   return result_info
     ? {
-        result,
+        result: await result,
```

Comment:
This is till required, it's a bug that we weren't `await`ing potentially async values here

Reply:
I'm worried typescript didn't catch it, but alas computers...